superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -71,12 +86,14 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
       newSliceName: props.sliceName,
       alert: null,
       action: this.canOverwriteSlice() ? 'overwrite' : 'saveas',
+      datasetName: '',
     };
     this.onDashboardSelectChange = this.onDashboardSelectChange.bind(this);
     this.onSliceNameChange = this.onSliceNameChange.bind(this);
     this.changeAction = this.changeAction.bind(this);
     this.saveOrOverwrite = this.saveOrOverwrite.bind(this);
     this.isNewDashboard = this.isNewDashboard.bind(this);
+    this.handleDatasetNameChange = this.handleDatasetNameChange.bind(this);
```

Comment:
This .bind is actually not needed because you used an arrow function to define handleDatasetNameChange function.  Arrow functions within a class component automatically bind this to the class component.  In other words you can remove line 96 completely.  If the other functions in this component were converted to arrow functions, then the binds from line 91-95 could also be remove :)